index.js

```diff
@@ -20,13 +20,27 @@ async function getProdUrl(sha) {
     throw new Error("Commit sha for prod url didn't match")
   }
 
+  const awaitBuild = core.getInput("await-build")
+
+  if(awaitBuild && data.deployments[0].state !== "READY"){
+    throw Error("Deployment not yet ready")
+  }
+
+
   return data.url
 }
 
 async function getBranchUrl(sha) {
-  const url = `https://api.vercel.com/v5/now/deployments?meta-githubCommitSha=${sha}`
+  const teamId = core.getInput("team-id")
+  const url = `https://api.vercel.com/v5/now/deployments?${teamId ? `teamId=${teamId}&` : ""}meta-githubCommitSha=${sha}`
   const { data } = await axios.get(url, headers)
 
+  const awaitBuild = core.getInput("await-build")
+
+  if(awaitBuild && data.deployments[0].state !== "READY"){
+    throw Error("Deployment not yet ready")
+  }
```

Comment:
This logic could be moved to a helper function to be used in both places.

Reply:
Updated in ec5b21c

Thank you for the initial work, this action has been really useful!

Reply:
Glad to hear it has been useful!  There is nothing quite like the feeling of building an open source projects that other people find useful!